pkg/notewriter/notewriter.go

```diff
@@ -0,0 +1,45 @@
+package notewriter
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+type NoteWriter struct {
+	investigationName string
+	sb                strings.Builder
+	logger            *zap.SugaredLogger
+}
+
+func New(investigationName string, logger *zap.SugaredLogger) *NoteWriter {
+	nw := &NoteWriter{investigationName, strings.Builder{}, logger}
+	nw.sb.WriteString(fmt.Sprintf("🤖 Automated %s pre-investigation 🤖\n", investigationName))
+	nw.sb.WriteString("===========================\n")
+	return nw
+}
+
+func (n *NoteWriter) String() string {
+	return n.sb.String()
+}
+
+func (n *NoteWriter) writeWithLog(format string, a ...any) {
```

Comment:
praise: i love that we start logging it all that was a rather important step missing

Reply:
And on top of that, it's optional: no logger passed, no logging! :)